core/control-plane/contract-core/src/main/java/org/eclipse/edc/connector/contract/offer/ContractOfferResolverImpl.java

```diff
@@ -38,25 +38,33 @@
 import java.time.ZonedDateTime;
 import java.util.Optional;
 import java.util.concurrent.atomic.AtomicLong;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import static java.lang.Math.max;
 import static java.lang.Math.min;
+import static java.util.stream.Collectors.toList;
 import static java.util.stream.Stream.concat;
 
 /**
  * Implementation of the {@link ContractOfferResolver}.
  */
 public class ContractOfferResolverImpl implements ContractOfferResolver {
+    private final String participantId;
```

Comment:
just as an idea: maybe there could be a method `getOwnParticipantId()` or similar on the `ParticipantService` instead of the explicit `participantId` argument?

Reply:
There is one on `ServiceExtensionContext`, but `ParticipantAgentService` is for foreign participants, not the current runtime.